app/controllers/buses_controller.rb

```diff
@@ -0,0 +1,49 @@
+class BusesController < ApplicationController
+  before_action :find_bus_by_id, only: %i[edit update destroy]
+
+  def index
+    @buses = Bus.all.order('id')
+  end
+
+  def new
+    @bus = Bus.new
+  end
+
+  def edit; end
+
+  def create
+    @bus = Bus.new(bus_params)
+    if @bus.save
+      redirect_to action: 'index', status: :created
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @bus.update(bus_params)
+      redirect_to action: 'index', status: :see_other
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @bus.destroy
+    redirect_to action: 'index', status: :see_other
+  end
+
+  private
+
+  def bus_params
+    params.require(:bus).permit(:code, :bus_type, :company, :capacity)
+  end
+
+  def find_bus_by_id
+    @bus = Bus.find_by(id: params[:id])
+    return if @bus
+
+    flash[:alert] = "Bus #{t(:not_found)}"
```

Comment:
We normally do do use i18n like this. `Bus not found` this whole line should be under translation. Don't you think in Japanese Bus will be called in a different way?
Try to use` i18n.t()` format for the messages.

Reply:
Thanks for the feedback apu. Used `I18n.t()`.